src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1314,19 +1314,19 @@ event::ptr primitive_inst::execute(const std::vector<event::ptr>& events) {
     _mem_changed = false;
     const auto orig_outputs = _outputs;
     std::vector<event::ptr> dependencies;
-    if (is_dynamic() && !has_inner_networks()) {
+    if ((is_dynamic() || _node->is_in_shape_of_subgraph()) && !has_inner_networks()) {
         do_runtime_in_place_concat();
         OPENVINO_ASSERT(_node != nullptr, "[GPU] Invalid primitive_inst object for dynamic shapes case: program_node can't be null");
         update_shape();
 
-
         bool can_skip_execution = false;
         if (_impl_params->output_layouts[0].count() == 0) {
             GPU_DEBUG_TRACE_DETAIL << id() << " : Skipping because output data is empty " << std::endl;
             can_skip_execution = true;
         }
 
-        if (_node->is_in_shape_of_subgraph()) {
+        // subgraph_input_changed can be available only shape_of is dynamic.
+        if (_node->is_in_shape_of_subgraph() && dependant_shape_of_insts.front()->is_dynamic()) {
```

Comment:
In my guess, this condition is added because if the dependent_shape_of_insts is static, then there is no way to check the shape_of_subgraph is executed at least once. Right? If so could you please elaborate more on the comments for future handling?

Reply:
Yes, static shape_of could run every-time even though no shape changed. But frankly it doesn't happen in real model because of constant propagation in ngraph. Anyway I'll add comment static case could be run every run.

Reply:
Applied it.